### PR TITLE
feat(build): added webpack plugin to be able to deploy in production

### LINKED ISSUE #8: Add webpack plugin for updating API url based on current environment
Will use replace-webpack-plugin

### PR DESCRIPTION
**Pull request by rjtbansal**
cc: @tristanMatthias 
Along with what I missed, I would like to know how I could test this before we could merge. :) 

Fixes #8

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -21,6 +24,11 @@ module.exports = withTM({
         loader: 'react-svg-loader'
       }
     );
+    cfg.plugins.push(new replacePlugin({
+      values: {
+        '%%IS_PROD%%': isProd ? 'https://codementoring.co/' : 'http://localhost:3000'
```
Comment (tristanMatthias) [inline next.config.js:29]: Can you change `%%IS_PROD%%` to `%%API%%` please, as this is what the actual value is :)

Comment (rjtbansal) [reply next.config.js]: Fixed the urls and using it in API.ts

File: src/lib/api.ts [hunk 2]
```diff
@@ -3,9 +3,10 @@ import axios from 'axios';
 
 export abstract class API {
 
-  static url = 'http://localhost:4000/graphql';
+  static url = '%%API%%/graphql';
 
   static async signup(user: UserInput) {
+    console.log(this.url);
```
Comment (tristanMatthias) [inline src/lib/api.ts:9]: Remove this

Comment (rjtbansal) [reply src/lib/api.ts]: Sorry, my bad. Done :)

**Pull request merged**